src/shared/watchdog.c

```diff
@@ -70,6 +71,26 @@ static int get_pretimeout_governor(char **ret_gov) {
         return read_virtual_file(sys_fn, WATCHDOG_GOV_NAME_MAXLEN, ret_gov, NULL);
 }
 
+static int set_pretimeout_governor(const char *governor) {
+        _cleanup_free_ char *sys_fn = NULL;
+        int r;
+
+        if (strempty(governor))
+                return 0; /* Nothing to do */
+
+        r = get_watchdog_sysfs_path("pretimeout_governor", &sys_fn);
+        if (r < 0)
+                return r;
+
+        log_info("Watchdog: setting pretimeout_governor to '%s' via '%s'", governor, sys_fn);
+
+        r = write_string_file(sys_fn, governor, WRITE_STRING_FILE_DISABLE_BUFFER);
```

Comment:
maybe add `WRITE_STRING_FILE_VERIFY_ON_FAILURE`? might be nice to suppress errors here in case the string already matches expectations? also `WRITE_STRING_FILE_VERIFY_IGNORE_NEWLINE`